readers/timescale/messages.go

```diff
@@ -42,8 +42,8 @@ func (tr timescaleRepository) ReadAll(chanID string, rpm readers.PageMetadata) (
 
 	// If aggregation is provided, add time_bucket and aggregation to the query
 	if rpm.Aggregation != "" {
-		q = fmt.Sprintf(`SELECT EXTRACT(epoch FROM time_bucket('%s', to_timestamp(time/1000000))) *1000000 AS time, %s(value) AS value FROM %s WHERE %s GROUP BY 1 ORDER BY time DESC LIMIT :limit OFFSET :offset;`, rpm.Interval, rpm.Aggregation, format, fmtCondition(rpm))
-		totalQuery = fmt.Sprintf(`SELECT COUNT(*) FROM (SELECT EXTRACT(epoch FROM time_bucket('%s', to_timestamp(time/1000000))) AS time, %s(value) AS value FROM %s WHERE %s GROUP BY 1) AS subquery;`, rpm.Interval, rpm.Aggregation, format, fmtCondition(rpm))
+		q = fmt.Sprintf(`SELECT EXTRACT(epoch FROM time_bucket('%s', to_timestamp(time/1000000))) *1000000 AS time,publisher,protocol, %s(value) AS value FROM %s WHERE %s GROUP BY 1, publisher, protocol ORDER BY time DESC LIMIT :limit OFFSET :offset;`, rpm.Interval, rpm.Aggregation, format, fmtCondition(rpm))
+		totalQuery = fmt.Sprintf(`SELECT COUNT(*) FROM (SELECT EXTRACT(epoch FROM time_bucket('%s', to_timestamp(time/1000000))) AS time,publisher,protocol, %s(value) AS value FROM %s WHERE %s GROUP BY 1, publisher, protocol) AS subquery;`, rpm.Interval, rpm.Aggregation, format, fmtCondition(rpm))
```

Comment:
Do we need `GROUP BY` for `COUNT` ?

Reply:
Yes, we do need the `GROUP BY` for the query to be complete. Without it the request returns an error. However we do not need the 1 that follows the `GROUP BY`.

Reply:
we don't need to return the other values in count since they are not being utilized. For the `totalQuery`, let it remain as it was before